src/Neo/SmartContract/Native/CryptoLib.BLS12_381.cs

```diff
@@ -134,13 +272,210 @@ public static InteropInterface Bls12381Pairing(InteropInterface g1, InteropInter
                 G1Projective g => new(g),
                 _ => throw new ArgumentException("BLS12-381 type mismatch")
             };
+            EnsureG1PointValid(in g1a);
```

Comment:
This change needs a mainnet states compatibility check from @superboyiii because it changes the existing CryptoLib API.

Reply:
It probably needs a hardfork guard in this case.